tests/test_postgres_incremental_replication.py

```diff
@@ -128,84 +98,203 @@ def setUp(self):
                 our_inet       inet,
                 our_cidr       cidr,
                 our_mac        macaddr,
-                our_money        money)
-                """.format(canonicalized_table_name(test_schema_name, test_table_name, cur))
+                our_money      money)
+                """.format(db_utils.canonicalized_table_name(cur, test_schema_name, test_table_name))
 
                 cur.execute(create_table_sql)
 
-                #insert fixture data 1
-                our_ts = datetime.datetime(1997, 2, 2, 2, 2, 2, 722184)
+                # insert fixture data and track expected records
+                self.inserted_records = []
+                self.expected_records = []
+
                 nyc_tz = pytz.timezone('America/New_York')
+                our_time_offset = "-04:00"
+
+                # record 1
+                our_ts = datetime.datetime(1977, 3, 3, 3, 3, 3, 733184)
+                our_ts_tz = nyc_tz.localize(our_ts)
+                our_time  = datetime.time(10,9,8)
+                our_time_tz = our_time.isoformat() + our_time_offset
+                our_date = datetime.date(1964, 7, 1)
+                my_uuid =  str(uuid.uuid1())
+                self.inserted_records.append({
+                    'our_varchar' : "our_varchar 2",
+                    'our_varchar_10' : "varchar_10",
+                    'our_text' : "some text 2",
+                    'our_integer' : 44101,
+                    'our_smallint' : 2,
+                    'our_bigint' : 1000001,
+                    'our_decimal' : decimal.Decimal('9876543210.02'),
+                    quote_ident('OUR TS', cur) : our_ts,
+                    quote_ident('OUR TS TZ', cur) : our_ts_tz,
+                    quote_ident('OUR TIME', cur) : our_time,
+                    quote_ident('OUR TIME TZ', cur) : our_time_tz,
+                    quote_ident('OUR DATE', cur) : our_date,
+                    'our_double' : decimal.Decimal('1.1'),
+                    'our_real' : decimal.Decimal('1.2'),
+                    'our_boolean' : True,
+                    'our_bit' : '1',
+                    'our_json' : json.dumps({'nymn' : 77}),
+                    'our_jsonb' : json.dumps({'burgers' : 'good++'}),
+                    'our_uuid' : my_uuid,
+                    'our_citext' : 'cyclops 2',
+                    'our_store' : 'dances=>"floor",name=>"betty"',
+                    'our_cidr' : '192.168.101.128/25',
+                    'our_inet': '192.168.101.128/24',
+                    'our_mac' : '08:00:2b:01:02:04',
+                })
+                self.expected_records.append({
+                    'our_decimal': decimal.Decimal('9876543210.02'),
+                    'OUR TIME': '10:09:08',
+                    'our_text': 'some text 2',
+                    'our_bit': True,
+                    'our_integer': 44101,
+                    'our_double': decimal.Decimal('1.1'),
```

Comment:
why are we expecting float values to be decimals?  I see this isn't new, but it is odd.

Reply:
Looking into this now.

Reply:
Figure out IRL that we are doing this to make sure the tap doesn't lose precision of decimals (and floats) and we are testing that they are being sent as the right data type by testing schema